--- a/data/de.haeckerfelix.Fragments.appdata.xml.in	2019-09-13 17:21:03.867898336 +0200
+++ b/data/de.haeckerfelix.Fragments.appdata.xml.in	2019-09-13 17:21:44.865126022 +0200
@@ -34,7 +34,7 @@
   <url type="bugtracker">https://gitlab.gnome.org/World/Fragments/issues</url>
   <url type="donation">https://de.liberapay.com/haecker-felix</url>
   <url type="translate">https://hosted.weblate.org/engage/fragments/</url>
-
+  <content_rating type="oars-1.0" />
   <kudos>
     <kudo>HiDpiIcon</kudo>
     <kudo>ModernToolkit</kudo>
